### ISSUE TITLE
abnormal data lead to (almost) dead loop

### ISSUE THREAD

**Issue by waynexia** [GreptimeTeam/greptimedb#2023]
### What type of bug is this?

Unexpected error

### What subsystems are affected?

Datanode

### What happened?

For parquets with range [`(0, 1690189123)`, `(0, 1)`], it will lead to a dead loop on `align_time_spans_to_windows`:

https://github.com/GreptimeTeam/greptimedb/blob/fcff66e03904d80aacb91b8edd4e15240161d264/src/storage/src/window_infer.rs#L126-L135

This loop will loop from 0 to 1690189123 with step 1.

### What operating system did you use?

doesn't matter

### How can we reproduce the bug?

create two parquet with time range [`(0, 1690189123)`, `(0, 1)`]

**Comment by v0y4g3r** [comment]
We need to update the `min_duration_to_window_size` function to accommodate durations with a difference of two orders of magnitude. That is to say, we allow a SST/memtable can be splitted into **at most 100 windows**.

**Comment by xxaier** [comment]
In the absence of abnormal data, this sql took 31 seconds (about dozens of data in the database)
在没有异常数据的情况下，这个sql花了31秒（数据库大概几十条数据）

<img width="1230" alt="image" src="https://github.com/GreptimeTeam/greptimedb/assets/130831741/3e85b472-b809-48a4-a020-8b264d8f7de7">

**Comment by v0y4g3r** [comment]
@xxaier With the fix in #2033 we can now cmplete the SQL normally:

```sql
mysql> SELECT CAST(ts as BIGINT) t,cid,rid FROM seen WHERE uid=1 AND ts>ARROW_CAST(0,'Timestamp(Millisecond,None)') ORDER BY ts LIMIT 8192;
+---------------+------+--------+
| t             | cid  | rid    |
+---------------+------+--------+
| 1690266985976 |    2 | 163984 |
| 1690266985977 |    2 | 289000 |
| 1690266985978 |    2 | 163362 |
| 1690266985979 |    2 | 270974 |
| 1690266985980 |    2 | 211851 |
| 1690266986998 |    2 | 250111 |
| 1690266986999 |    2 | 275699 |
| 1690266987000 |    2 | 219988 |
| 1690266987001 |    2 | 480591 |
| 1690266987002 |    2 | 210044 |
| 1690266987003 |    2 |  56924 |
| 1690266987996 |    2 | 481396 |
| 1690266987997 |    2 | 481393 |
| 1690266987998 |    2 | 480574 |
| 1690266987999 |    2 | 257786 |
| 1690266988000 |    2 | 260078 |
| 1690268934915 |    2 | 303804 |
| 1690268934916 |    2 | 266205 |
| 1690268934917 |    2 | 481522 |
| 1690268934918 |    2 | 481603 |
| 1690268934919 |    2 | 185763 |
| 1690268935911 |    2 | 291956 |
| 1690268935912 |    2 | 481584 |
| 1690268935913 |    2 | 268629 |
| 1690268935914 |    2 | 176401 |
| 1690268935915 |    2 | 107862 |
| 1690268936690 |    2 | 481682 |
| 1690268936691 |    2 | 250112 |
| 1690268937708 |    2 |   9575 |
| 1690268937709 |    2 | 128463 |
| 1690268937710 |    2 | 481600 |
| 1690268937711 |    2 | 181461 |
| 1690268937712 |    2 | 181204 |
| 1690268937713 |    2 | 276997 |
| 1690268938712 |    2 |  16010 |
| 1690268938713 |    2 | 481657 |
| 1690268938714 |    2 | 278295 |
| 1690268938715 |    2 |   8079 |
| 1690268938716 |    2 | 481761 |
| 1690268938717 |    2 | 288590 |
| 1690269417876 |    2 | 481330 |
| 1690269417877 |    2 | 380269 |
| 1690269417878 |    2 | 480578 |
| 1690269417879 |    2 | 480576 |
| 1690269417880 |    2 | 275698 |
| 1690269417881 |    2 | 265547 |
| 1690269605721 |    2 | 163983 |
| 1690269605722 |    2 | 169920 |
| 1690269605723 |    2 | 481384 |
| 1690269605724 |    2 | 250630 |
| 1690269605725 |    2 | 167860 |
| 1690271607306 |    2 | 480580 |
| 1690271607307 |    2 | 117046 |
| 1690271607308 |    2 | 260133 |
| 1690271607309 |    2 | 250113 |
| 1690271607310 |    2 |  47464 |
+---------------+------+--------+
56 rows in set (0.44 sec)
```
